src/pc/tracker.rs

```diff
@@ -473,7 +519,10 @@ impl SubxtTracker {
 					.count() as u32;
 				let voted_against = dispute_info.statements.len() as u32 - voted_for;
 
-				let misbehaving_validators = if voted_for > voted_against {
+				let outcome =
+					if voted_for > voted_against { SubxtDisputeResult::Valid } else { SubxtDisputeResult::Invalid };
```

Comment:
shouldn't the outcome be  based on events ?

Reply:
At this point we might not see the corresponding event, as it comes later.

Reply:
I think we should only take the outcome from the events when it comes later and not rely on these calculations.

Reply:
The main problem is that the result from the events comes later, so at that point we cannot find misbehaving validators, which is a useful functionality. In fact, this fix should be postponed till we have the proper collector service. In that case, we can gather disputes information for a single candidates from multiple inherent datas and join it together. For now, I see no other way to preserve partial data between multiple blocks. And I'm inclined not to remove this code completely as it is very useful to find misbehaving validators. Even if it is not now 100% correct, it is better to fix it once we can keep and share data across multiple blocks.

Reply:
Gotcha, makes sense.